cmd/nodecmd/local.go

```diff
@@ -242,3 +263,265 @@ func notImplementedForLocal(what string) error {
+RPC URL of the L1. 
+
+This command can only be used to validate Proof of Stake L1.`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: localValidate,
+	}
+
+	cmd.Flags().StringVar(&blockchainName, "l1", "", "specify the blockchain the node is syncing with")
+	cmd.Flags().StringVar(&blockchainName, "blockchain", "", "specify the blockchain the node is syncing with")
+	cmd.Flags().Uint64Var(&stakeAmount, "stake-amount", 0, "(PoS only) amount of tokens to stake")
+	cmd.Flags().StringVar(&rpcURL, "rpc", "", "connect to validator manager at the given rpc endpoint")
+	cmd.Flags().Uint64Var(&balance, "balance", 0, "set the AVAX balance of the validator that will be used for continuous fee on P-Chain")
+	cmd.Flags().Uint16Var(&delegationFee, "delegation-fee", 100, "(PoS only) delegation fee (in bips)")
+	cmd.Flags().StringVar(&aggregatorLogLevel, "aggregator-log-level", constants.DefaultAggregatorLogLevel, "log level to use with signature aggregator")
+	cmd.Flags().BoolVar(&aggregatorLogToStdout, "aggregator-log-to-stdout", false, "use stdout for signature aggregator logs")
+
+	return cmd
+}
+
+func localValidate(_ *cobra.Command, args []string) error {
+	clusterName := ""
+	if len(args) > 0 {
+		clusterName = args[0]
+	}
+	if ok, err := node.CheckClusterIsLocal(app, clusterName); err != nil || !ok {
+		return fmt.Errorf("local cluster %q not found, please create it first using avalanche node local start %q", clusterName, clusterName)
+	}
+
+	network, err := networkoptions.GetNetworkFromCmdLineFlags(
+		app,
+		"",
+		globalNetworkFlags,
+		true,
+		false,
+		localStartSupportedNetworkOptions,
+		"",
+	)
+	if err != nil {
+		return err
+	}
+
+	fee := network.GenesisParams().TxFeeConfig.StaticFeeConfig.AddSubnetValidatorFee
+	kc, err := keychain.GetKeychainFromCmdLineFlags(
+		app,
+		"to pay for transaction fees on P-Chain",
+		network,
+		keyName,
+		useEwoq,
+		useLedger,
+		ledgerAddresses,
+		fee,
+	)
+	if err != nil {
+		return err
+	}
+
+	// should take input prior to here for stake amount, delegation fee, and min stake duration
+	if stakeAmount == 0 {
+		stakeAmount, err = app.Prompt.CaptureUint64Compare(
+			"Enter the amount of token to stake",
+			[]prompts.Comparator{
+				{
+					Label: "Positive",
+					Type:  prompts.MoreThan,
+					Value: 0,
+				},
+			},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	if rpcURL == "" {
+		rpcURL, err = app.Prompt.CaptureURL("What is the RPC endpoint?", false)
+		if err != nil {
+			return err
+		}
+	}
+	_, blockchainID, err := utils.SplitAvalanchegoRPCURI(rpcURL)
+	if err != nil {
+		return err
+	}
+	chainSpec := contract.ChainSpec{
+		BlockchainID: blockchainID,
+	}
+
+	if balance == 0 {
+		availableBalance, err := utils.GetNetworkBalance(kc.Addresses().List(), network.Endpoint)
+		if err != nil {
+			return err
+		}
+		balance, err = blockchain.PromptValidatorBalance(app, availableBalance/units.Avax)
+		if err != nil {
+			return err
+		}
+	} else {
+		// convert to nanoAVAX
```

Comment:
this has changed, please check latest main on how to best do this for float avax balance